src/example.ts

```diff
@@ -1,115 +0,0 @@
-    type: 'GetRandomGif',
-    topic: string,
-    onFail: (x: TaskError<string>) => FetchErrorAction,
-    onSuccess: (a: TaskSuccess<string>) => FetchSuccessAction,
-}
-type Task = GetRandomGifTask;
-
-type MainState = {
-    status: 'not started' | 'pending' | 'success' | 'error',
-    result: Option<Result<string, string>>
-}
-type State = {
-    main: MainState
-};
-const patch = <O, P>(o: O, p: P): O & P => Object.assign({}, o, p);
-const reducer = (state: MainState, action: Action): [MainState, Option<Task[]>] => {
-    switch (action.type) {
-        case ActionTypes.Fetch:
-            return [patch(state, { status: 'pending', result: None }), new Some([
-                create<GetRandomGifTask>({
-                    type: 'GetRandomGif',
-                    topic: 'food',
-                    onFail: createFetchErrorAction,
-                    onSuccess: createFetchSuccessAction,
-                })
-            ])];
-        case ActionTypes.FetchSuccess:
-            return [patch(state, { status: 'success', result: new Some(action.result) }), None]
-        case ActionTypes.FetchError:
-            return [patch(state, { status: 'error', result: new Some(action.result) }), None]
-        default:
-            return [state, None];
-    }
-}
-
-const initialState: State = {
-    main: {
-        status: 'not started',
-        result: None
-    }
-};
-
-const createGifUrl = (topic: string): string => `https://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC&tag=${topic}`;
-const decodeGifUrl = (response: any): string => response.data.image_url;
-// Callbacks must be on Task because otherwise we don't know their type
-// Is result of Task compaitable with callback? We only know this if
-// they are coupled
-const myTaskRunner: TaskRunner<Action> = <X, A>(task: Task): Promise<Action> => {
-    switch (task.type) {
-        case 'GetRandomGif':
-            const url = createGifUrl(task.topic);
-            return fetch(url)
-                .then(response => response.json())
-                .then(decodeGifUrl)
-                .then(createSuccess)
-                .catch(createError)
-                .then(result => (
-                    result.success
-                        ? task.onSuccess(result)
-                        : task.onFail(result)
-                ))
-    }
-}
-
-const store = enhancedCreateStore(myTaskRunner, combineReducers<State, Task>({ main: reducer }), initialState);
-
-const rootEl = document.getElementById('root');
-store.subscribe(() => {
-    const state = store.getState();
-    if (rootEl) rootEl.innerHTML = `
-<pre>
-Status: ${state.main.status}
-Result success: ${state.main.result
-    .map(result => result.success)
-    .getOrElse(() => false)}
-Result success value/failure reason: ${state.main.result
-    .map(result => result.success ? JSON.stringify(result.value, null, '\t') : result.value)
-    .getOrElse(() => '')}
-</pre>`;
-});
-
-store.dispatch({ type: ActionTypes.Fetch });
```

Comment:
Keen to keep this example, if it's not hard to update to the new format (just `Observable.of` in the task runner?). Will use it one day as a tutorial of sorts. If not we can revive it later.